bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ThingStatusTriggerHandler.java

```diff
@@ -77,6 +79,9 @@ public ThingStatusTriggerHandler(Trigger module, BundleContext bundleContext) {
             this.types = Set.of(ThingStatusInfoChangedEvent.TYPE);
         }
         this.bundleContext = bundleContext;
+
+        this.eventTopicFilter = Pattern.compile("^openhab/things/" + thingUID.replace("*", ".*?") + "/.*$");
+
         Dictionary<String, Object> properties = new Hashtable<>();
         properties.put("event.topics", "openhab/things/" + thingUID + "/*");
```

Comment:
What is this used for? I didn't find code that uses this property. It also contradicts the description [here](https://www.openhab.org/docs/developer/utils/events.html#receive-events) which states that regular expressions should be used as filters.

Reply:
I did the initial implementation based on the [GenericEventTriggerHandler](https://github.com/openhab/openhab-core/blob/9eef6a3b6d2a75514b5495e8400c2712b66bdb68/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/GenericEventTriggerHandler.java#L48) which uses nearly the same code for event subscription. Currently I cannot tell you whether there is a spacial purpose or not.